querybook/webapp/ui/StyledText/StyledText.tsx

```diff
@@ -0,0 +1,220 @@
+        props.cursor
+            ? ` cursor: ${props.cursor ? props.cursor : 'inherit'}`
+            : ''};
+    ${(props) =>
+        props.hover
+            ? `  
+                &:hover {
+                    color: var(--text-hover);
+                }`
+            : ''};
+    ${(props) =>
+        props.untitled
+            ? `  
+                opacity: 0.7;
+                font-style: italic;`
+            : ''};
+    ${(props) =>
+        props.center
+            ? `  
+                display: flex;
+                justify-content: center;`
+            : ''};
+
+    * {
+        ${(props) =>
+            props.weight
+                ? ` font-weight: ${
+                      props.weight === 'bold'
+                          ? 'var(--bold-font)'
+                          : `var(--${props.weight}-bold-font)`
+                  }`
+                : ''};
+        ${(props) =>
+            props.size
+                ? ` font-size: ${
+                      props.size === 'text'
+                          ? 'var(--text-size)'
+                          : `var(--${props.size}-text-size)`
+                  }`
+                : ''};
+        ${(props) =>
+            props.color
+                ? ` color: ${
+                      props.color === 'text'
+                          ? 'var(--text)'
+                          : props.color.startsWith('accent')
+                          ? `var(--color-${props.color})`
+                          : `var(--text-${props.color})`
+                  }`
+                : ''};
+        ${(props) =>
+            props.accent
+                ? `
+                        font-family: var(--font-accent); 
+                        letter-spacing: var(--letter-spacing${
+                            props.weight === 'bold' || props.weight === 'extra'
+                                ? '-bold'
+                                : ''
+                        })
+                        `
+                : ''};
+        ${(props) =>
+            props.hover
+                ? `  
+                    &:hover {
+                        color: var(--text-hover);
+                    }`
+                : ''};
+    }
+`;
+
+export const StyledText: React.FunctionComponent<IProps> = ({
+    children,
+    className,
+    tooltip,
+    tooltipPos = 'up',
+    accent,
+    noUserSelect,
+    ...elementProps
+}) => {
+    const textClassName = clsx({
+        Text: true,
+        [className]: Boolean(className),
+    });
+
+    if (tooltip) {
+        elementProps['aria-label'] = tooltip;
+        elementProps['data-balloon-pos'] = tooltipPos;
+    }
+
+    elementProps['accent'] = Boolean(accent);
+    elementProps['noUserSelect'] = Boolean(noUserSelect);
+
+    return (
+        <StyledTextSpan {...elementProps} className={textClassName}>
+            {children}
+        </StyledTextSpan>
+    );
+};
+
+export const AccentText: React.FunctionComponent<IProps> = ({
+    children,
+    className,
```

Comment:
className doesn't need to be explicitly stated? since you can auto include it in elementProps